app/src/main/java/com/babylon/wallet/android/presentation/settings/securitycenter/securityfactors/composables/FactorSourcesList.kt

```diff
@@ -0,0 +1,112 @@
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.babylon.wallet.android.R
+import com.babylon.wallet.android.designsystem.composable.RadixSecondaryButton
+import com.babylon.wallet.android.designsystem.theme.RadixTheme
+import com.babylon.wallet.android.presentation.dialogs.info.GlossaryItem
+import com.babylon.wallet.android.presentation.ui.composables.InfoButton
+import com.babylon.wallet.android.presentation.ui.composables.card.FactorSourceCardView
+import com.babylon.wallet.android.presentation.ui.model.factors.FactorSourceCard
+import com.radixdlt.sargon.FactorSourceId
+import kotlinx.collections.immutable.PersistentList
+
+@Composable
+fun FactorSourcesList(
+    modifier: Modifier = Modifier,
+    mainFactorSource: FactorSourceCard?,
+    factorSources: PersistentList<FactorSourceCard>,
+    @StringRes factorSourceDescriptionText: Int,
+    @StringRes addFactorSourceButtonTitle: Int,
+    onFactorSourceClick: (FactorSourceId) -> Unit,
+    onAddFactorSourceClick: () -> Unit,
+    onInfoClick: (GlossaryItem) -> Unit,
+) {
+    LazyColumn(
+        modifier = modifier.fillMaxWidth(),
+        contentPadding = PaddingValues(horizontal = RadixTheme.dimensions.paddingDefault),
+        verticalArrangement = Arrangement.spacedBy(RadixTheme.dimensions.paddingDefault)
+    ) {
+        item {
+            Text(
+                modifier = Modifier.padding(
+                    top = RadixTheme.dimensions.paddingDefault,
+                    bottom = RadixTheme.dimensions.paddingLarge
+                ),
+                text = stringResource(id = factorSourceDescriptionText),
+                style = RadixTheme.typography.body1HighImportance,
+                color = RadixTheme.colors.gray2
+            )
+        }
+        mainFactorSource?.let {
+            item {
+                Text(
+                    modifier = Modifier.padding(bottom = RadixTheme.dimensions.paddingDefault),
+                    text = stringResource(id = R.string.factorSources_list_default),
+                    style = RadixTheme.typography.body1HighImportance,
+                    color = RadixTheme.colors.gray2
+                )
+                FactorSourceCardView(
+                    modifier = Modifier
+                        .padding(bottom = RadixTheme.dimensions.paddingMedium)
+                        .clickable { onFactorSourceClick(it.id) },
+                    item = it
+                )
+            }
+            item {
```

Comment:
This item has to be shown only if there are other factor sources.

Reply:
true good catch